app/javascript/components/Study.js

```diff
@@ -1,57 +1,103 @@
+/* eslint-disable require-jsdoc */
 import React from 'react'
 
-const descriptionWordLimit = 100
+const descriptionWordLimit = 750
+const summaryWordLimit = 150
+const lengthOfHighlightTag = 21
 
 /* converts description into text snippet */
-export function formatDescription(rawDescription) {
+export function formatDescription(rawDescription, term) {
   const textDescription = stripTags(rawDescription)
-  return shortenDescription(textDescription)
+  return shortenDescription(textDescription, term)
 }
 
-// return the first 100 words, with a '...' appended if needed'
-// this returns a node, not text, so we can italicize 'continued'
-function shortenDescription(textDescription) {
-  const wordArray = textDescription.split(' ')
-  const numWords = wordArray.length
-  // take the first 100 words, or the whole thing if shorter
-  const shortenedText = wordArray.slice(0, descriptionWordLimit - 1)
-                                 .join(' ')
-
-  let suffixTag = <></>
-  if (numWords > descriptionWordLimit) {
-    suffixTag = <span className="detail"> ...(continued)</span>
+function highlightText(text, terms) {
+  const matchedIndicies = []
+  if (terms) {
+    let match
+    const regex = RegExp(terms, 'gi')
+    // Find indices where match occured
+    while ((match = regex.exec(text)) != null) {
+      matchedIndicies.push(match.index)
+    }
+    if (matchedIndicies.length>0) {
```

Comment:
the code would be simpler (although slightly less performant), if you dropped this length check, and just always returned { styledText: text.replace(regex, `<span id='highlight'>${terms}</span>`), matchedIndicies }. (even if there are no terms)

Reply:
This won't work due to there being multiple terms. If there are no matches there's no need to to iterate over termMatches to replace text